_pages/home/release-notes/checkout-libraries-release-notes.md

```diff
@@ -0,0 +1,395 @@
+      this.eventService
+        .get(CheckoutDeliveryAddressCreatedEvent)
+        .subscribe(({ userId, cartId }) => {
+          this.globalMessageService.add(
+            { key: 'addressForm.userAddressAddSuccess' },
+            GlobalMessageType.MSG_TYPE_CONFIRMATION
+          );
+
+          this.eventService.dispatch(
+            { userId, cartId },
+            CheckoutResetDeliveryModesEvent
+          );
+
+          this.eventService.dispatch({}, CheckoutResetQueryEvent);
+        })
+    );
+    ...
+```
+
+`CheckoutResetDeliveryModesEvent` is fired to re-load a cart, which ends up being caught by our `CheckoutLegacyStoreEventListener`. This event listener encapsulates the ngrx actions that we can not remove from checkout until other features changes to commands and queries.
+
+The example for `CheckoutResetQueryEvent` being fired is to reset our super query from `CheckoutQueryFacade` due to the nature of our commands and queries, where we can [reload or reset a query](https://sap.github.io/spartacus-docs/commands-and-queries/#queries-overview).
+
+### How to properly make use of queries in checkout
+
+Example: [CheckoutQueryFacade](https://github.com/SAP/spartacus/blob/develop/feature-libs/checkout/base/core/facade/checkout-query.service.ts) and [CheckoutDeliveryAddressComponent](https://github.com/SAP/spartacus/blob/develop/feature-libs/checkout/base/components/checkout-delivery-address/checkout-delivery-address.component.ts)
+
+The 'super query' that is injected in other checkout 'facade' services is used to get the current checkout details, which contains information on `delivery address, delivery mode, and payment information` when using base checkout. When using b2b checkout, you will have additional data like `purchase order number`, `cost centers`, and `payment type`.
+
+```typescript
+  protected checkoutQuery$: Query<CheckoutState | undefined> =
+    this.queryService.create<CheckoutState | undefined>(
+      () =>
+        this.checkoutPreconditions().pipe(
+          switchMap(([userId, cartId]) =>
+            this.checkoutConnector.getCheckoutDetails(userId, cartId)
+          )
+        ),
+      {
+        reloadOn: this.getQueryReloadTriggers(),
+        resetOn: this.getQueryResetTriggers(),
+      }
+    );
+
+    getCheckoutDetailsState(): Observable<QueryState<CheckoutState | undefined>> {
+      return this.checkoutQuery$.getState();
+    }
+```
+
+The getCheckoutDetailsState returns the entire state, which is an object composed of loading, error, and data properties, but you can always create additional methods in your extended service to include only getting the data. You would only need to use `.get()` instead of `.getState()` when executing queries.
+
+```typescript
+    getCheckoutDetails(): Observable<CheckoutState | undefined> {
+      return this.checkoutQuery$.get();
+    }
+```
+
+## How to add additional logic to existing features
+
+TBD (wrappers / new schematics mechanism)
```

Comment:
Hmm, maybe just point to the doc here. 
Instead of repeating ourselves for all features, we can just point to the general doc which explains how to do it

Reply:
which doc? 

I was going to add the docs after our schematics gets updated to support the wrapper modules

Reply:
Ah, got it.
I was thinking about this doc: https://sap.github.io/spartacus-docs/lazy-loading-guide/#customizing-lazy-loaded-modules
I guess we will extract this example to a separate doc page.

Reply:
I found trails to this referred docs

https://github.com/SAP/spartacus/blob/develop/projects/schematics/README.md#glossary > https://github.com/SAP/spartacus/pull/15237 > https://wiki.one.int.sap/wiki/display/spar/Installing+extensions+for+an+existing+lazy+loaded+feature

btw, in our repo, it says to notify to update the link in this file, so letting you know now to update haha~

----
All in all, 

shouldn't this 'docs' be made available in the public spartacus `docs` repo as well, so that we can reference it here, which is why we have this TBD pending

Reply:
Thanks for letting me know!
Yep, agreed. Until then, I linked to our existing doc which also explains how to extend LL features (if curious, please check the commits).